Random zeds/ferals have no loot

### Describe the bug

Not consistently, but both @gettingusedto and I have noticed that some new zombies and even ferals have no loot when killed, even if they're newly seen and probably (definitely, in the case of ferals) not revived.

### Attach save file

[Vanilla-trimmed.tar.gz](https://github.com/user-attachments/files/29362361/Vanilla-trimmed.tar.gz)

This save has a bunch of ferals, but I couldn't reproduce the bug just spawning ferals and then killing them. 

### Steps to reproduce

1) Kill zombies until some don't drop loot.

From my testing, debug-spawned zombies always drop loot, so something is happening elsewhere in the chain.

### Expected behavior

Newly-killed zombies that didn't revive still have all their loot, and ferals especially always have all their loot.

### Screenshots

<img width="1202" height="381" alt="Image" src="https://github.com/user-attachments/assets/ba48b7f6-0f62-4097-9cdf-3f648c704961" />

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.26100.8655 (24H2)
- Game Version: cdda-experimental-2026-06-21-0506-dirty+SDL3 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms]
]


### Additional context

_No response_

Unfortunately that change doesn't compile. I've got a bunch of errors like

`error C2665: 'JsonValue::read': no overloaded function could convert all the argument types`

--- a/src/game.cpp
+++ b/src/game.cpp
@@ -4938,7 +4938,6 @@ bool game::revive_corpse( const tripoint_bub_ms &p, item &it, int radius )
         return false;
     }
 
-    critter.death_drops = false;
     critter.add_effect( effect_downed, 5_turns, true );
 
     if( it.get_var( "no_ammo" ) == "no_ammo" ) {

--- a/src/savegame_json.cpp
+++ b/src/savegame_json.cpp
@@ -2627,7 +2627,11 @@ void monster::load( const JsonObject &data )
     }
     data.read( "mission_fused", mission_fused );
     // for migration, remove in 0.K
-    data.read( "no_extra_death_drops", death_drops );
+    if( data.has_object( "no_extra_death_drops" ) ) {
+        bool no_extra_death_drops;
+        data.read( "no_extra_death_drops", no_extra_death_drops );
+        death_drops = !no_extra_death_drops;
+    }
     data.read( "death_drops", death_drops );
     data.read( "spawn_corpse", spawn_corpse );
     data.read( "death_message", death_message );

